src/test/java/org/mule/service/http/impl/service/client/GrizzlyHttpsClientProxyTestCase.java

```diff
@@ -0,0 +1,127 @@
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.mule.rules.BouncyCastleProviderCleaner;
+import org.mule.runtime.api.tls.TlsContextFactory;
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.HttpClientConfiguration;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.mule.runtime.http.api.domain.message.response.HttpResponseBuilder;
+import org.mule.runtime.http.api.server.HttpServerConfiguration;
+import org.mule.service.http.impl.functional.client.AbstractHttpClientTestCase;
+import org.mule.tck.http.TestProxyServer;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.tck.junit4.rule.SystemProperty;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Issue;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jsse.provider.BouncyCastleJsseProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class GrizzlyHttpsClientProxyTestCase extends AbstractHttpClientTestCase {
+
+  private static final String PASS = "mulepassword";
+
+  @Rule
+  public DynamicPort proxyPort = new DynamicPort("proxyPort");
+  @Rule
+  public SystemProperty useProxyProperties = new SystemProperty(ASYNC_CLIENT + "useProxyProperties", "true");
+  @Rule
+  public SystemProperty proxyHostProperty = new SystemProperty("http.proxyHost", "localhost");
+  @Rule
+  public SystemProperty proxyPortProperty = new SystemProperty("http.proxyPort", proxyPort.getValue());
+
+  private TestProxyServer proxyServer = new TestProxyServer(proxyPort.getNumber(), port.getNumber(), false);
+  private HttpClient client;
+
+  @Rule
+  public BouncyCastleProviderCleaner bouncyCastleProviderCleaner = new BouncyCastleProviderCleaner();
+
+  public GrizzlyHttpsClientProxyTestCase(String serviceToLoad) {
+    super(serviceToLoad);
+  }
+
+  private void ensureBouncyCastleProviders() {
+    insertProviderAt(new BouncyCastleProvider(), 1);
+    insertProviderAt(new BouncyCastleJsseProvider(), 2);
+  }
+
+  @Before
+  public void createClient() throws Exception {
+    ensureBouncyCastleProviders();
+    proxyServer.start();
+    client = service.getClientFactory().create(new HttpClientConfiguration.Builder()
+        .setTlsContextFactory(TlsContextFactory.builder()
+            .trustStorePath("tls/trustStore")
+            .trustStorePassword(PASS)
+            .build())
+        .setName("httpsProxyHeader-test")
+        .build());
+    client.start();
+  }
+
+  @After
+  public void stopClient() throws Exception {
+    if (proxyServer != null) {
+      proxyServer.stop();
+    }
+    if (client != null) {
+      client.stop();
+    }
+  }
+
+  @Override
+  protected HttpResponse setUpHttpResponse(HttpRequest request) {
+    HttpResponseBuilder response = HttpResponse.builder();
```

Comment:
verify here that no incorrect headers are passed.

Reply:
Assertion added.